src/main/java/core/basesyntax/strategy/OperationStrategyImpl.java

```
@@ -0,0 +1,16 @@
+package core.basesyntax.strategy;
+
+import java.util.Map;
+
+public class OperationStrategyImpl implements OperationStrategy {
+    private final Map<FruitTransaction.Operation, OperationHandler> operationHandlers;
+
+    public OperationStrategyImpl(Map<FruitTransaction.Operation,
+            OperationHandler> operationHandlers) {
+        this.operationHandlers = operationHandlers;
+    }
+
+    public OperationHandler getOperationHandler(FruitTransaction.Operation operation) {
+        return operationHandlers.get(operation);
```

Consider adding a check to handle cases where the `operationHandlers` map does not contain a handler for the given operation. You might want to throw an exception or return a default handler in such cases.